cpp/src/arrow/compute/exec/expression.h

```diff
@@ -112,7 +113,7 @@ class ARROW_EXPORT Expression {
 
     // post-bind properties
     ValueDescr descr;
-    int index;
+    internal::SmallVector<int, 2> indices;
```

Comment:
Is `SmallVector`'s second parameter a lower bound?

Reply:
It's the number of items that can be stored before it changes from a stack-based to heap-based representation.

Reply:
Got it, thanks.